Environment changes to support multicluster e2e tests.

john-a-joyce opened the pull request.

tiswanso committed:
Multicluster e2e test infrastructure

These changes are the initial infrastructure changes to enable multi-cluster
e2e tests.  This change will read the cluster registry config file and
create multiple kube client interfaces to access both the local and remote
clusters

john-a-joyce committed:
Multicluster e2e test infrastructure
These changes are the initial infrasturcture changes to enable multicluster
e2e tests. This change will read the cluster registry config file and
create  multiple kube client interfaces to access both the local and remote
clusters.  It will also enable the Pilot manifests to start up using
the cluster registry.

rcurran1 commented on pilot/test/util/kubernetes.go:
```diff
@@ -38,28 +38,60 @@ const (
+					"istio-injection": injectionValue,
+				},
 			},
-		},
```
The only difference between the if and else code is "GenerateName: prefix" and "Name: namespace". 
Anyone know how to update to pass that in as an object (code below is wrong/guess)? That would simplify this code block to:
if namespace == "" {
   keyName = "GenerateName:"
   valueName = prefix
} else {
   keyName = "Name:"
   valueName = namespace
}
ns, err := c1.....Create(...)

rcurran1 commented on tests/e2e/tests/pilot/util/environment.go:
```diff
@@ -163,21 +170,58 @@ func (e *Environment) ToTemplateData() TemplateData {
+			return err
+		}
+        }
+	// CLusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG environment
```
type "CLus..." -> "Clus..."

rcurran1 commented on tests/e2e/tests/pilot/util/environment.go:
```diff
@@ -163,21 +170,58 @@ func (e *Environment) ToTemplateData() TemplateData {
+		}
+        }
+	// CLusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG environment
+	// Variable.  The test config can be defined to use either a single cluster or 2 clusters
```
Is starting this comment w/ "Variable." some convention in this file?

john-a-joyce replied:
No convention - I fixed it up

rcurran1 commented on tests/e2e/tests/pilot/util/environment.go:
```diff
@@ -163,21 +170,58 @@ func (e *Environment) ToTemplateData() TemplateData {
+	if e.Config.ClusterRegistriesDir != "" {
+		e.clusterStore, err = clusterregistry.ReadClusters(e.Config.ClusterRegistriesDir)
+		if e.clusterStore == nil {
+			return err
```
I take it that just returning here w/ no messaging is OK. i.e. the calling function will provide enough info on the failure in Setup().

john-a-joyce replied:
I can add a log message.

rcurran1 commented on tests/e2e/tests/pilot/util/environment.go:
```diff
@@ -163,21 +170,58 @@ func (e *Environment) ToTemplateData() TemplateData {
+			return err
+		}
+		if e.clusterStore != nil {
+
```
blank line

kyessenov committed:
Pod cache should only track RUNNING and PENDING pods (#4145)

frankbu committed:
Bookinfo sample v1alpha3 routing rules (#4150)

* Bookinfo v1alpha3 routing rules

* temporary use apiVersion config.istio.io/v1alph2

* use new apiVersion: networking.istio.io/v1alpha3

* Declare all subsets up front

* add details:v2 subset

* address review comments

* address review comments

costinm committed:
Followup on EDS (#4036)

Automatic merge from submit-queue.

Followup on EDS

- reduce logging (use env variable until someone finds out how to configure modules in the new logger)
- add a fail-safe env variable, in case we need to turn it off. Needs to be set in the injector template if needed.
- add the TLS option for grpc port (first attempt - will replace with proper grpc listener in followup). 
Shriram: yes, I still agree we should load the certs in pilot, but can be done later.
- starting to add some structs for debug and push - in progress, will keep adding while the rest is reviewed. Main question is how to name the clients for the debug - based on POD ID (IP) and watched resource, pod name and resource, or  group by remote pod. A http handler listing all connected endpoints and the watched resources in followup.
- finish the move of all eds-related functions to eds.go (from mesh_discovery)

Also this changes the 2 'istio-on-istio' pilot configs to v2.

Commit:
Fix pilot tests to use www.google.com instead of nghttp2.org (#4221)

nghttp2.org is currently unreachable.

john-a-joyce committed:
Multicluster e2e test environment setup
These changes are the initial environment changes to enable multicluster
e2e tests. This change will read the cluster registry config file and
create  multiple kube client interfaces to access both the local and remote
clusters.  It will also enable the Pilot manifests to start up using
the cluster registry.

rkpagadala committed:
githubctl chmod in current directory (#4229)

Automatic merge from submit-queue.

githubctl chmod in current directory

frankbu committed:
Add missing route to delay rule (#4241)

loewenstein-sap committed:
Location of istioctl code changed (#4251)

Automatic merge from submit-queue.

Location of istioctl code changed

geeknoid committed:
Cleanup attribute bag usage in mixer/pkg/api. (#4228)

* Cleanup attribute bag usage in mixer/pkg/api.

- Use snapshotting feature to fix bug where Check was not returning the correct set of
referenced attributes for quota results. It wasn't included attributes referenced from
the APA.

- Get rid of the compatibility bag goo from the gRPC server code. This is no longer
needed.

* Convert remaining uses of target.* attributes to destination.* instead.

kyessenov committed:
Disable TestMixerInternalFail (#4267)

* Disable TestMixerInternalFail
follow up in #4264

costinm committed:
Add a connection number to the tracking (#4261)

* Add a connection number to the tracking, to deal with the strange out of order connect

* Lint

kyessenov committed:
disable it (#4268)

Signed-off-by: Kuat Yessenov <kuat@google.com>

guptasu committed:
Adapter integration test framework [has-vendor-changes] (#4132)

**Utility for all adapters to write baseline based mixer integration tests.**

Also added prometheus, redisquota and list adapter integration tests.

This has been a common ask from adapter devs, so adding few adapter devs in the review for feedback and input.

Features
======
* _Trigger parallel Mixer calls_ : Multiple async calls to adapter via Mixer.
* _Initial setup/teardown_: Allow test to do adapter specific initial setup (eg. start fake backend server) and teardown (do backend server cleanup). (see redisquotatest)
* _All purpose baseline section for adapter specific baseline data_: Test can specify any (interface{}) adapter specific data to be part of baseline. Example: for prometheus adapter, actual metric reported to backend can be embedded into the expected json baseline. (see prometheus test)
* _manipulate config and input attrs_: Test can easily manipulate config or call attributes to create test variety (see redisquotatest, list adapter)


Thanks @ozevren and @geeknoid for most of the ideas.

* Adapter integration test framework

* Cleanup

* update lock file

* Address feedback from Doug and Jeff

* changes after rebase

* Make linter happy

* Address feedback from Oz

* update vendor

* update lock after rebase

* update lock and vendor after rebase

* fix the test

* update vendor

chxchx committed:
Prow postsubmit runs unit tests sequentially (#4242)

* Postsubmit runs unit tests sequentially

* set -parallel 1

* revert presubmit unit tests

Commit:
Add secret "delete" permission to Istio CA. (#4259)

JimmyCYJ committed:
Wait until port is up before sending requests to Envoy proxy. (#4237)

* Wait until port is up before sending requests to Envoy proxy.

* Fix lint

* Fix lint error.

selmanj committed:
Support 0.6.0 (#4085)

Automatic merge from submit-queue.

Support Istio 0.6.0 using GCP+DM template

Adds support for istio 0.6.0 and drops support for 0.4.0.

kyessenov committed:
Disable TestCheckReportDisable (#4278)

Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit:
Refactor security/cmd/nodeagent_k8s/. (#4243)

This is for #4190 , summary of the refactor:

- Put the workload related package together under workload/, and use workload_app/ to distinguish application with istio logic. (workloadapi can also be put into workload/ later).
- Rename server to handler, reduce nesting level, use Options as constructor parameter.
- Change some class/symbols from public to private.
- Updates some comment.

kyessenov committed:
Disable v1alpha3 from pre-submit on circle and prow (#4258)

Commit:
Do not add secret back when service account is removed. (#4275)

qiwzhang committed:
Fix flaky CheckReportDisable test. (#4280)

qiwzhang committed:
Fix flaky mixer/test/client tests (#4274)

* Use different base-id for each Envoy restart in mixerclient tests.

* enable mixer_internal_fail_test

* Fix format

qiwzhang committed:
Remove mixer client stress test. (#4282)

kyessenov committed:
move crd test to e2e (#4283)

Signed-off-by: Kuat Yessenov <kuat@google.com>

jwendell committed:
Remove duplicate docker "scratch" directory. (#4269)

Automatic merge from submit-queue.

Remove duplicate docker "scratch" directory.

We already have ISTIO_DOCKER defined in main Makefile for
that purpose.

geeknoid committed:
Improve generated docs for commands. (#4187)

- Don't output a flag table if there are no flags for a command.

- Don't output a Shorthand column in a flag table if there are no shorthands defined for the
command.

- Attach mixc's tracing flags to the check and report commands instead of at the top since
that's where they apply.

john-a-joyce committed:
Multicluster e2e test environment setup

Addresses some comments on the initial PR as well as fixing the lint errors.

Commit:
Add interface and mock for CA client. (#4284)

diemtvu committed:
Use authentication policy to service-to-service mTLS. (#4089)

* Use authentication policy to service-to-service mTLS.

* Lint

* Remove pre-mature changes. authn policy doen't need to be part of buildListenerOpts yet.

* Remove unused definitions.

* Add e2e test case.

costinm committed:
CircleCI and test optimizations (#4232)

* Pass testing.T to pilot tests, for better reporting (will break down into subtests)

* Move egress to flaky

* Missed the target

* Updates

* Fixing reruning test

* Fix syntax

* Removing dependencies from lint/buid - it is indeed faster

* Test was timing out

* Fix the junit conversions. Revert failing codecov if unit test coverage changed

* Use test run for better reporting

* Fix vendor again

* Move mixer/dashboard to periodic

* Fix lint and actually report the error

* More tests converted

* Envoy dep is in istio.deps now

* Revert golang changes, waiting for Nate

guptasu committed:
Fix integration test to not cause port reuse issue. (#4299)

Automatic merge from submit-queue.

Fix integration test to not cause port reuse issue.

Setting port to 0. This is exactly what the old redisquota integration test was doing too.

kyessenov committed:
Update go-control-plane and istio/api (#4300)

ijsnellf committed:
No default routes in v1alpha3 (#4025)

* remove default routes

* fix model test

* fix e2e rule

mangchiandjjoe committed:
Improve the SDSServer features (#4120)

* Implemented DeregisterUdsPath

* Added data protection with Mutex

* Return same version number for the same X.509 key/cert

* Addressed code reviews

geeknoid committed:
Cleanup mixer/pkg/api and get coverage to 100% (#4277)

aaronshurley committed:
Adds copilot snapshot during cf config setup (#4019)

- cloudfoundry can now dynamically inject route rules
based on data from copilot
- during a snapshot, the config monitor will keep its config store in
tact if a connection error occurs while attempting to update itself.
- copilot e2e test: fix apiversion
- don't log error in monitor package
- copilot_snapshot: add deadline for querying routes
- we are now attempting to cache the routes in the copilot
snapshot. Needlessly creating route objects that may
already be there was bad.
- server boostrap has a pattern emerging now that
the two monitors and one controller have been extracted.
- change v3routing to networking everywhere

mandarjog committed:
Use dynamically assigned port for Prometheus integration tests (#4312)

kyessenov committed:
Disable cache tests (#4315)

Automatic merge from submit-queue.

Disable cache tests

Fixes flakes due to https://github.com/istio/istio/issues/4304.
Signed-off-by: Kuat Yessenov <kuat@google.com>

geeknoid committed:
Remove CODEOWNERS and update OWNERS files to match GitHub teams used with CODEOWNERS (#4316)

frankbu committed:
Fix istioctl defaultNamespace not initialized (#4310)

* Fix istioctl defaultNamespace not initialized

* gofmt

ijsnellf committed:
fix generated crd kind (#4305)

Automatic merge from submit-queue.

Fix generated CRD kind

The CRD kinds were accidentally using kabob case instead of camel case. So for example route rules had kind `route-rule` instead of `RouteRule`.

kyessenov committed:
Update istio/api (#4318)

SeriyBg committed:
Fix #4325 - istioctl get list of resources by default must scan all namespaces (#4326)

Automatic merge from submit-queue.

Fix #4325 - istioctl get list of resources by default must scan all namespaces

jmuk committed:
adds validation of referential integrity for mixer configs (#1963)

Automatic merge from submit-queue.

adds validation of referential integrity for mixer configs

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

rshriram committed:
Fix incorrect label match used to build Inbound HTTP routes (#4301)

Automatic merge from submit-queue.

Fix incorrect label match used to build Inbound HTTP routes

Removing incorrect code related to source label match, and adding the right set of checks.

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

yangminzhu committed:
Define CRDs for ServiceRole and ServiceRoleBinding in pilot (#4246)

Automatic merge from submit-queue.

Define CRDs for ServiceRole and ServiceRoleBinding in pilot

1. create corresponding CRD for ServiceRole and ServiceRoleBinding (See
https://github.com/istio/api/blob/master/rbac/v1alpha1/rbac.proto).
The ServiceRole and ServiceRoleBinding is currently defined in istio-auth.yaml
but we have plans to migrate them to pilot for the local RBAC.

2. add corresponding validation code for new model types (#3927).

3. in longer term after the local RBAC is implemented, the pilot watches for changes on Istio RBAC policies, fetches the updated RBAC policies from Istio Config Store and distributes relevant Istio RBAC policies to Envoy proxies that are co-located with service instances and enforce the policies inside Envoy (The mixer won't handle the RBAC policies anymore).

See the design doc (WIP) for more details: https://docs.google.com/document/d/1XkN9ls-ZDISz2uXGN20RaIre63DkJJOJNUOzcKCxbWA/edit?usp=sharing

burnettk committed:
allow cleanup.sh scripts to be re-run without outputing errors (#3617)

costinm committed:
Finally get rid of the dependencies and cleanup (#4313)

* Finally get rid of the dependencies, no longer used and slower, and better organize deps

* Add placeholder waiting for admin

* Forgot the defaults

Commit:
Add KeyCertBundleRotator for automatic cert rotation with the upstream CA. (#4295)

costinm committed:
Remove remaining dependecies of EDS to envoy/v1 (#4348)

mandarjog committed:
Add clusters and fix filenames to yaml (#4306)

GregHanson committed:
rename missed v1alpha2 references to v1alpha3 (#4328)

* rename missed v1alpha2 references to v1alpha3

* comment fixes

* lint fixes

ijsnellf committed:
Fix badly defined v1alpha3 test rule (#4327)

* fix badly defined rule

* fix version

* unbreak vendor

guptasu committed:
use the vendored protos from gogo instead of curling. (#4347)

* use the vendored protos from gogo.

plus update the istio/api.

guptasu committed:
Update istio/api (#4351)

guptasu committed:
update vendor of istio/istio to match vendor-istio (#4354)

Commit:
Add a util function for patching webhook config (#4294)

* Add util for patching webhook.

* Address reviewer comments.

* Refine error messages.

rshriram committed:
Remove some dead code (v1alpha1) from v2/ folder (#4350)

* Correctly build inbound route based on routes, mirror/redirect for v1alpha3

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* fixes

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* short comment

* cleaning up some references to v1alpha1 in v2 code

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* undo vendor

* nits

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* move servers out

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* restore v1alpha1 routing

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

GregHanson committed:
external service registry for registry aggregator (#4108)

* first pass external service registry for registry aggregator

* review comments

* bug fix

* switched to event handler for detecting changes to ExternalServices crds

* bug fix in tests

* include Resolution in conversion

* lint fixes

* lint fixes

* minor refactors

andraxylia committed:
Upgrade to new proxy to pick up Envoy cluster warming + Makefile fix (#4292)

* Upgrade to new proxy to pick up Envoy cluster warming

* Add dependency on istio.deps

* Workaround grpc cds failure

* Print the md5 of envoy used to start the local process

* pick-up fix for envoy.conf.template

* Re-trigger circleci

Commit:
IPv6: Proxy address validation (#4263)

Automatic merge from submit-queue.

IPv6: Proxy address validation

Modified logic to allow validation of IPv6 addresses, which will have the IP
address enclosed in square brackets.

ldemailly committed:
adding myself back

frankbu committed:
Fix tls settings validation (#4335)

Automatic merge from submit-queue.

Fix tls settings validation

Partial fix for #4105

diemtvu committed:
Port v1 authentication modules to v2 (#4352)

Automatic merge from submit-queue.

Port v1 authentication modules to v2

This ports [PR 4089](https://github.com/istio/istio/pull/4089), which change the logic to also consider authentication policy when configure listener/cluster SSL. The actual configuration is still left open (TODO @mostrowski)

Issue: https://github.com/istio/istio/issues/4337

rshriram committed:
pilot: Separate networking/api code from grpc server code (#4356)

* Correctly build inbound route based on routes, mirror/redirect for v1alpha3

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* fixes

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* short comment

* cleaning up some references to v1alpha1 in v2 code

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* undo vendor

* nits

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* move servers out

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* restore v1alpha1 routing

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* separate networking and xds server logic

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* lint

* more duplicate code removal

* undo del of xds.go

* lint

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* more nits

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* lint

* rebase for authn PR

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

xiaolanz committed:
Turn on racetest for all presubmits (#3780)

* make racetest mandatory for all presubmits

* disable redis race test

* exclude security e2e

* make it after test

diemtvu committed:
Move authentication.go to model. (#4371)

ijsnellf committed:
Set integer seconds for CORS max age (#4324)

* set max age as seconds

* fix vendor (hopefully)

rshriram committed:
Data plane and plugin interfaces for better decoupling from xds (#4367)

* data plane interfaces

* lint

* moving networking into dataplane

* tweaks

* nits

* lint

* moving back to networking

* renaming/comments

chxchx committed:
Remove serial execution contraint on postsubmit unit tests (#4379)

Automatic merge from submit-queue.

Remove serial execution contraint on postsubmit unit tests

Such constaints were imposed due to a bug from go-junit-report. Remove them now that bug has been resolved.

fixes https://github.com/istio/istio/issues/4230

ldemailly committed:
[vendor-change] Fortio update to 0.8.0 - with semver not need to specify the tag, dep picks up the latest release tag (#4383)

Automatic merge from submit-queue.

[vendor-change] Fortio update to 0.8.0 - with semver not need to specify the tag, dep picks up the latest release tag

2 phase commit needed when tests pass

changes/release notes:
https://github.com/istio/fortio/releases/tag/v0.8.0

vadimeisenbergibm committed:
Refactor e2e bookinfo tests (#4185)

* extract bookinfoSampleDir constant

* delete rulesDir, let the routing rules reside in multiple directories

* added /kube prefix to the route rule paths

* copy the rules together with their directories into rulesDir

* os.MkDirAll -> os.MkdirAll

* filepath.Dir(rule) -> filepath.Dir(dest)

* append bookinfoSampleDir constant in the code

* util.GetResourcePath -> getBookinfoResourcePath

left over from the previous PR

* remove leading '/' from the yamls, add it to the bookinfoSampleDir

* remove yaml extension from the constants, add it programmatically

* replaced filepath.Join() with String concatenation

* remove redundant '/'

will be added by filepath.Join()

* added deploymentDir and routeRulesDir constant

* append routeRulesDir as a constant

* add yamlExtension in apply/delete Rules

* add missing yamlExtension

jmuk committed:
remove bin/perftest.sh (#4374)

Automatic merge from submit-queue.

remove bin/perftest.sh

This is not used, and bin/perfcheck.sh seems more preferable.
This PR also adds some baselines for benchmark tests.

Fixes #3869

ijsnellf committed:
set the resolution and mesh external flags of the model service in k8s, cf, consul and eureka (#4393)

guptasu committed:
IBP handle service interface (#4353)

Automatic merge from submit-queue.

IBP handle service interface

Added code generation for template specific IBP service `HandleXXXXService`.

**Additional changes**
* Running doc generation on Augmented template instead of original meta template proto.
* Renamed augmented proto files `*_instance.proto` to `*_handler_service.proto`.
* Fix tons of comments on the generated augmented proto.

**Here is a peek of how the generated docs now look:**
[sample_generated_doc.zip](https://github.com/istio/istio/files/1822061/sample_doc.zip)

**How to start reviewing:** 
1. Start looking at the generated augmented protos under `mixer/template/**/*_handler_service.proto`. 
2. Look at all the *handler_service.proto under mixer/template, they all might have varying structure in the meta templates (sub messages or not, check/report/quota), which affects the generated protos.

**Questions:**
1. I have create a new message called `InstanceMsg` (Instance with suffix "Msg")that is proto version of our go `Instance` struct. So, IBP will get `InstanceMsg` (`InstanceMsg` struct for go flavor of IBP) while LAP will get `Instance` struct. Does it seem confusing ?

**NOTE:** (PR looks huge, but code change is only inside tools/codegen/pkg, other stuff is generated.)

Commit:
Allow webhook to load certs dynamically. (#4329)

Allow webhook to load certs dynamically.

guptasu committed:
Turns out mixer_codegen was not including the generated gRPC service block. Fixed (#4396)

guptasu committed:
[vendor-change] update istio/api  (#4382)

Automatic merge from submit-queue.

[vendor-change] update istio/api 

istio/api no longer contains old istio/mixer/v1/cfg.proto and…v1/descriptor/value_type.proto.

vendor https://github.com/istio/vendor-istio/pull/32

Commit:
NodeAgent send CSR to CA and writes cert into file. (#4365)

In NodeAgent server.go, when WorkloadAdded invoked, send CSR request to CA; save the response to file by SecretServer. Currently the server.go implementation lives in node_agent/, will merge with node_agent/na/nodeagent.go later.

ostromart committed:
LDSv2 fixes (#4377)

* LDSv2 fixes

* Restore docker.envoy2 target, remove debug code

* Make changes in deprecated/ also in v1alpha2/

* Restore lost v2 config file

* Move LDSConnection back to v2

* Remove pretty package

costinm committed:
Few more fixes to circle - nightly was broken (#4399)

GregHanson committed:
plug in the external service registry by default to the aggregator (#4397)

* plug the external service registry by default

* lint fixes

* lint fix

* lint fix

* lint

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

ldemailly committed:
vendor update for fortio and api 3/20/2018 matching https://github.com/istio/vendor-istio/pull/35 (#4410)

sebastienvas committed:
Adds a script for v1alpha3 (#4407)

Automatic merge from submit-queue.

Adds a script for v1alpha3

Needs https://github.com/istio/test-infra/pull/742
Fixes #4270

rkpagadala committed:
fix broken proxy docker (#4412)

Automatic merge from submit-queue.

fix broken proxy docker build

for regular (non debug) builds envoy should come from ISTIO_ENVOY_RELEASE_PATH
instead it was dependent on $ISTIO_OUT/envoy which is wrong.

Cleaned up proxy_debug target to make it similar to proxy. Makes it easier to fix, if we need changes in the future.

fixes #4266

geeknoid committed:
Skip stuff we shouldn't include in our code coverage numbers. (#4400)

diemtvu committed:
Rerun generate_config_crd_types.go (#4417)

Automatic merge from submit-queue.

Rerun generate_config_crd_types.go to update ServiceRole and ServiceRoleBinding types.

Change https://github.com/istio/istio/pull/4246 might use the old script when generate types for the new CRDs.

ldemailly committed:
Update perf script and doc - better default run_tests (#4416)

* Update perf script and doc - better default run_tests

run_tests now run ing->f1 ing->f2 f1->f2 f2->f1 and saves the json and
starts report ui (if fortio is installed)

Simplify some of the setup and doc

Use 443 as the http port on the VM (should not get auto cleaned up)

Only run the through istio ingress test by default

Don’t delete everything by default (when not sourcing)

Fortio 0.8.x doesn’t need percentiles passed

* Typo

* Fix vm install to rebuild after go get with the -X taking action

* Fix ingress to f2 label

* More readme updates

* Fix up readme

* Change labels

diemtvu committed:
Migrate to authentication policy v1alpha2 (#4415)

* Switch to authentication policy v1alpha2.

* Complete authn policy version switch to v1alpha2

* Fix e2e.

* Change port.Match to use authn PortSelector directly.

geeknoid committed:
Fix newly broken links. (#4422)

Commit:
Removing Istio Release Tag Version from Default Variable (#4364)

Automatic merge from submit-queue.

[Ansible Install] Removing Istio Release Tag Version from defaults 

- If you have a version on default, the ``determine_latest_version.yml`` task will not work even if the extra var is determined with -e

GregHanson committed:
first pass v1alpha3 cds (#4403)

* first pass v1alpha3 cds

* refactoring + starting inbound cluster logic

* modified cluster name building

* Initial, very basic, not really working CDS v2

(cherry picked from commit 57e61bbc3e07331276a272986d3d586c986ffe25)

* removed unnecessary deprecated/clusters.go and references

* lint fixes

* lint fixes

Commit:
Fix data racing issues. (#4427)

yangminzhu committed:
Add ServiceRole and ServiceRoleBinding to istioctl CRD definition. (#4428)

Automatic merge from submit-queue.

Add ServiceRole and ServiceRoleBinding to istioctl CRD definition.

The CRD definition was added to pilot in #4246.

douglas-reid committed:
Improve dashboard test robustness (#4423)

xiaolanz committed:
remove fmt.sh (#4425)

Automatic merge from submit-queue.

remove fmt.sh

Keep it consistent with "make format"

diemtvu committed:
Adding Jwt filter to sidecar listener if it is needed for authentication. (#4343)

Automatic merge from submit-queue.

Adding Jwt filter to sidecar listener if it is needed for authentication.

Pilot will add Jwt filter to listener if it detect that Jwt is required by authentication policy. It also add the corresponding cluster for JwksURI (so that Jwt filter can make request to fetch public key for validation).

This PR use the existing Jwt filter implementation in Istio proxy. It may need to change to the upstream version when https://github.com/envoyproxy/envoy/issues/2514 complete.

Issue: https://github.com/istio/istio/issues/4337
Related: https://github.com/istio/istio/issues/1090

ldemailly committed:
Simplify and make e2e simple simple again/more readable (#4426)

* Simplify and make e2e simple simple again/more readable

* Linter

* Fix bug in timeout and code review comments

* Code review comments

Change the pod readiness retries to be faster

wattli committed:
Include pem in build package for better user experience (#4439)

Commit:
Move CA client config to pkg. (#4436)

bianpengyuan committed:
Add two e2e tests for mixer check cache (#4141)

Automatic merge from submit-queue.

Add two e2e tests for mixer check cache

Covers ingress and service sidecar cases. Part of fix for #2160

guptasu committed:
Make the supported templates public. (#4441)

* Make the supported templates public.

Commit:
Move CAClient interface defintion to the consumer KeyCertBundlerRotator. (#4440)

This makes adding methods to CAClient easier, no need to modify the fakeCAClient.

Commit:
Add upgrade test to post-submit test suite. (#4375)

Automatic merge from submit-queue.

Add upgrade test to post-submit test suite.

Commit:
Fix lint. (#4453)

Automatic merge from submit-queue.

Fix a small lint error

Commit:
Add an option for CA to append DNS to cert for webhook injector service (#4216)

Add an append-dns option to CA which allows CA to append DNS entries as alternative names for the webhook services.

Commit:
Move DetermineEnv logic to platform client. (#4454)

ayj committed:
istioctl: import all client auth plugins (#4438)

fixes https://github.com/istio/istio/issues/4083

rshriram committed:
Uniform cluster name for v1alpha3 (#4411)

* nits

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* nits

* minor code cleanup

* more tests

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* fix bad syntax

JimmyCYJ committed:
Test v1 config (#4408)

Automatic merge from submit-queue.

Test v1 config

Refactor test framework, and test that we support for v1 config.
Fixes# [1217](https://github.com/istio/proxy/issues/1217#event-1529883524)

costinm committed:
Fix eds reconnect bug (#4449)

guptasu committed:
Update istio/istio to new istio/api (#4474)

tiswanso committed:
Prow script for pilot-multicluster-e2e test

Runs multicluster e2e test.  Will use mason to get GKE resources in
the same GCP project.  The resource_type will need to be updated
after the test-infra commit of the boskos config change to create the new
resource_type for multiple GKE clusters.

Script can be run with pre-allocated clusters by having a pre-setup
kubeconfig for multiple clusters.  It will choose the pilot cluster to
be the first cluster in the get-contexts or the PILOT_CLUSTER env var.

example:
PILOT_CLUSTER=gke_istio-ci-testing_us-east1-c_cluster-1 \
  KUBE_USER=tiswanso@cisco.com ROOT=$(pwd) USE_MASON_RESOURCE=False \
  prow/istio-pilot-multicluster-e2e.sh

geeknoid committed:
Stop depending on the evaluator.IL type. (#4470)

- mixs's validation logic now depends on evaluator.TypeChecker instead
of evaluator.IL. This will let us delete a bunch of stuff when it comes
time to nuke mixer/pkg/runtime in a subsequent PR.

- Delete support for specifying the IL evaluator's default cache size.
This again is in support of soon deleting the IL evaluator as a whole.

jwendell committed:
Check earlier if Istio source code is in the right place (#4462)

Thus we get a more informative error message. For example,
if one clones Istio in their $HOME and try to run make:

Before:
$ make
Checking that Gopkg.* are in sync with vendor/ submodule:
if this fails, 'make pull' and/or seek on-call help
diff Gopkg.toml vendor/
diff Gopkg.lock vendor/
mkdir: cannot create directory ‘//out’: Permission denied
make: *** [//out/linux_amd64/release] Error 1

After this patch:
$ make
Istio not found in GOPATH/src/istio.io. Make sure to clone Istio on that path.
make: *** [Makefile:185: check-tree] Error 1

JimmyCYJ committed:
Allocate new ports for Envoy after restart. (#4465)

Automatic merge from submit-queue.

Allocate new ports for Envoy after restart.

Fixes #4444

mandarjog committed:
Kubernetes yaml diff tool (#4450)

* Add yaml diff tool

* Review comments

* spec can be None

* Update return code to be positive

diemtvu committed:
Change authn policy back to v1apha1. (#4485)

liminw committed:
Fix a bug in Istio RBAC code on prefix/suffix matching. (#4489)

Automatic merge from submit-queue.

Fix a bug in Istio RBAC code on prefix/suffix matching.

tiswanso committed:
Adress review comments and fix HUB error w/ pilot e2e

yangminzhu committed:
Add e2e tests for Istio RBAC. (#4349)

* Add e2e tests for Istio RBAC in mixer.

* Add Istio RBAC e2e tests.

* Fix lint.

* Change copyright to 2018.

* Address comments.

diemtvu committed:
Fix missing version change to v1alpha1 (#4497)

Automatic merge from submit-queue.

Fix missing version change to v1alpha1

Issues: 
https://github.com/istio/istio/issues/4471
https://github.com/istio/istio/issues/4330

ostromart committed:
LDSv2 fixes for sidecar (#4492)

Commit:
Fix the missing retry times and increase the max duration for upgrade test. (#4472)

* Increase the max duration for upgrade test.

* Fix lint.

* Add retry times.

* fix lint.

diemtvu committed:
Add circuitbreakers and ssl for jwks cluster v2. (#4446)

chxchx committed:
Portability of make e2e_all on mac (#4443)

Automatic merge from submit-queue.

Portability of make e2e_all on mac

Once this PR is in, please merge this one https://github.com/istio-releases/daily-release/pull/479

yutongz committed:
Copy start_envoy and envoy.conf.template into istio/istio from proxy (#4358)

Automatic merge from submit-queue.

Copy start_envoy and envoy.conf.template into istio/istio from proxy

fixes https://github.com/istio/istio/issues/4317

tiswanso committed:
Fix fmt and lint errors

geeknoid committed:
Remove races from cache tests (#4442)

* Remove races from cache tests

This expands the scope of a lock in the async evicter to satisfy the race test logic. It's not
strictly necessary, but it's better to keep the tools happy.

* Deal with another racy test.

john-a-joyce committed:
Changes required due to rebase

Add new paramater to KubeApply in RBAC test
Address review comments

ijsnellf committed:
Convenience tool for converting v1alpha1 configs to v1alpha3 (#4308)

Automatic merge from submit-queue.

Convenience tool for converting v1alpha1 configs to v1alpha3

From the command description:
> Converts sets of v1alpha1 configs to v1alpha3 equivalents on a best effort basis. The output should be considered a starting point for your v1alpha3 configs and probably require some minor modification.  Warnings will (hopefully) be generated where configs cannot be converted perfectly, or in certain edge cases.  The input must be the set of configs that would be in place in an environment at a given time.  This allows the command to attempt to create and merge output configs intelligently. Output configs are given the namespace and domain of the first input config so it is recommended that input configs be part of the same namespace and domain.

I had to recreate this PR after recreating my Istio fork. There are only minor differences between this and the old PR: config metadata is grabbed from the first config and a couple small fixes were made.

Note the following potential TODOs that may be handled in a future PR:
```
	// TODO: k8s ingress -> gateway?
	// TODO: create missing destination rules/subsets?
```

Contributes to https://github.com/istio/istio/issues/3621.

yangminzhu committed:
Update the Istio RBAC e2e test to use full name in services. (#4513)

geoand committed:
Install latest version of Istio by default (#3602)

Commit:
IPv6: Resolving host name to IP (#4369)

Automatic merge from submit-queue.

IPv6: Resolving host name to IP

The ResolveAddr() method takes a host:port and returns IP:port, where host can
be a hostname to be "resolved" or an IP address. This PR is another small fix
to the istio infrastructure to support IPv6 addresses, providing [ipv6]:port,
in that case. AFAICT, this is currently only used for statsd UDP address.

Some special considerations...

In the original code, net.LookupIP() can return multiple addresses, of which
the logic uses the first address. I tested this API with some hosts that have
both IPv4 and IPv6 addresses, and the IPv6 addresses were always second. If
this cannot be guaranteed, then additional logic would need to be added to
specify, say, IPv4 priority for the value used.

To support IPv6 only mode of operation, the host provided must either be an
IPv6 address, or a host name that (only) resolves to an IPv6 address. Note:
Kubernetes does not support dual-stack operation as of this time.

This change does not include altering the help string for the statsd UDP
address, which can be done as a separate commit and should trigger a doc
update, once IPv6 support is fully operational. Currently, this and other
commits are trying to get the infrastructure support in place for IPv6.

Fixes issue #4096 
/area networking

geeknoid committed:
Delete mixer/pkg/runtime, long live runtime2

geeknoid committed:
Rename mixer/pkg/runtime2 to mixer/pkg/runtime (#4518)

geeknoid committed:
Bump up coverage numbers in a few packages. (#4519)

guptasu committed:
Add registry to consume templates and adapters. (#4455)

Automatic merge from submit-queue.

Add registry to consume templates and adapters.

Creates a single registry for ingestion of adapter info and base64 encoded template descriptors. PR also contains provision for ingesting template descriptors that are not tied to any adapters (example: all the core built-in templates).

This PR provides the wiring needed for next work of taking descriptor and yaml config to generate bytes.

**NOTE:** This PR depend on [new Info proto pr](https://github.com/istio/api/pull/426) 

100% codecoverage woohoo!!

costinm committed:
Consistency improvements, fixes and docs for v2 (#4475)

* Better fix and cleanup on v2

* Improvements and consistency for the v2 handlers, and a readme

* Switch to CDSv2 and add the v2/alhpa3 proxy image to docker images

* Reduce debug for AZ, more debug and cleanup

* Starting of a small test for cds

* Attempt to fix private fork builds

* Make sure mixer can override -p setting

guptasu committed:
100% codecov in model.go

wattli committed:
Add .pem to install/kubernetes for better user experience (#4517)

rshriram committed:
CDSv2 fixes, and parity with v1alpha3 (#4515)

* CDSv2 fixes, and parity with v1alpha3

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* nits

* trim

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* format

christian-posta committed:
Let's include the helper template files for helm. Fixes https://github.com/istio/istio/issues/3895 (#4467)

Automatic merge from submit-queue.

Let's include the helper template files for helm.

geeknoid committed:
Refactor mixer/pkg/expr & mixer/pkg/il (#4521)

Automatic merge from submit-queue.

Refactor mixer/pkg/expr & mixer/pkg/il

We now have mixer/pkg/lang that holds the expression language stuff, and mixer/pkg/il which is now
language-agnostic.

There's no code change here, just moving stuff around and adjusting imports accordingly.

Commit:
Add better NewKeyCertBundleRotator for easier initialization of keycertbundlerotator. (#4464)

* Add rotatorfactory for easier initialization of keycertbundle.

* Fix based on comments.

* Small fix.

xiaolanz committed:
fix flaky race condition in TestCrdsRetryMakeSucceed (#4505)

Automatic merge from submit-queue.

fix flaky race condition in TestCrdsRetryMakeSucceed

addresses #4486 

All tests in this file set retryInterval=0 in  getTempClient()  that causes a race condition when tests are running in parallel.  

Also exclude a api server test from racetest.

Commit:
IPv6: Updating parsing of host/port for bootstrap options. (#4525)

Automatic merge from submit-queue.

IPv6: Updating parsing of host/port for bootstrap options.

Allow IPv6 addresses (with square brackets) to be specified in
addresses that are parsed into host/port for various bootstrap
file arguments.

Added unit tests to exercise the logic, which has been placed into
functions for use by multiple options. Tests cover failure cases,
which were not previously checked. Rely on the existing golden
master to test the happy path for all entries. Didn't feel we needed
to add a golden master for IPv6 addressess, as covered by the UTs.

/area networking
Fixes #4097 issue

Commit:
IPv6: Generate error, if no address provided for resolution (#4526)

Automatic merge from submit-queue.

IPv6: Generate error, if no address provided for resolution

Not an IPv6 specific item, but doing as cleanup to PR4369. First change is
to return an error, if no address is specified for IP resolution. Previously,
the code would not return an error, but instead, just return an empty string
as the result. However, the ONLY place that currently uses this, is for statsd
UDP address, where the caller checks for empty strying first.  Changing to
raise an error (which makes more sense), and if future callers need to ignore
this case, they can check against the new error definition.

The second change is to remove duplication in lookup of IP address.
Follow-up PR for Issue #4096 
/area networking

rshriram committed:
Use v1alpha3 for LDSv2 (#4496)

* convert LDSv2 to use v1alpha3

* tweak

* lint

* fix incorrect cluster generation

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* Upstream TLS Settings

* dummy filter for orig_dst listener

* lint

* remove unnecessary fields in envoy struct

* more trimming

* fix

* nits

* remove cds/rds

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* nits

* nits

* lint

* tweaks and tests

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* format

* more tests

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* add legacy ingress support

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* mgmt ports

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

liminw committed:
Fix Istio RBAC samples to use fully qualified service names. (#4509)

guptasu committed:
Pipe template name into generated code. (#4511)

Automatic merge from submit-queue.

Pipe template name into generated code.

Change codegen to publish the `template_name` into augmented template.
Use the `template_name` from augmented template when ingesting template in registry.go.

christian-posta committed:
Update hub variable for helm when doing updateVersion.sh. Fixes https://github.com/istio/istio/issues/4476 (#4478)

Automatic merge from submit-queue.

Update hub variable for helm when doing updateVersion.sh.

Fixes #4476

Commit:
Reuse CAClient.CreateRequest for NodeAgent server. (#4477)

* Reuse CAClient.CreateRequest for NodeAgent server.

- refactoring CAClient to support fetching keycert for
multiple identity.
- KeyCertBundle adds a CertOptions method for CAClient to invoke.

* Fix lint

* Fix the stopped mutex

* Print ids content instead

* lint and fix

* Remove failed test in rotator_test.go will add back in OnPrem_test.go later

* remove unnecesssary mixer change.

vadimeisenbergibm committed:
Refactor test version routing (#4387)

Automatic merge from submit-queue.

Refactor test version routing

To enable more routing by version tests, to remove code duplication, to make the tests table-driven.

rshriram committed:
Split CircleCI tasks into smaller ones (#4529)

* circleci cleanups

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

* tweaks

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

bianpengyuan committed:
Deflake check cache e2e test (#4495)

* Deflake check cache test by making large number of calls in short time.

* lint.

Commit:
match mixer service selectors to the pod spec in the deployment and the rest of the components (#4522)

Automatic merge from submit-queue.

[helm] Mixer Service Selectors Do Not Match Pod Spec 

When using the latest helm chart on master the mixer Service [pod selectors](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/service.yaml#L28-L31) don't match the mixer [pod spec](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/mixer/templates/deployment.yaml#L17-L18) in the deployment.

Other services only select based on the [`istio` label](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/pilot/templates/service.yaml#L27-L28) so I changed the mixer service to match.

### Example yaml from `helm template`

```yaml
---
# Source: istio/charts/mixer/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: istio-mixer
  labels:
    app: mixer
    chart: mixer-0.5.0
    release: test-template
    heritage: Tiller
    istio: mixer
spec:
  ports:
  - name: tcp-plain
    port: 9091
  - name: tcp-mtls
    port: 15004
  - name: http-health
    port: 9093
  - name: configapi
    port: 9094
  - name: statsd-prom
    port: 9102
  - name: statsd-udp
    port: 9125
    protocol: UDP
  - name: prometheus
    port: 42422
  selector:
    app: istio-mixer
    release: test-template
    istio: mixer

---
# Source: istio/charts/mixer/templates/deployment.yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: istio-mixer
  labels:
    app: mixer
    chart: mixer-0.5.0
    release: test-template
    heritage: Tiller
    istio: mixer
  annotations:
    checksum/config-volume: aa9a748156816384e60a0e3a3ed8ac8f8fa44a41fe50127558bcd09379cc03b5
spec:
  replicas: 1
  template:
    metadata:
      labels:
        istio: mixer
      annotations:
        sidecar.istio.io/inject: "false"
    spec:
      serviceAccountName: istio-mixer-service-account
      containers:
        - name: statsd-to-prometheus
          image: "prom/statsd-exporter:latest"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 9102
          - containerPort: 9125
            protocol: UDP
          args:
          - '-statsd.mapping-config=/etc/statsd/mapping.conf'
          resources:
            {}

          volumeMounts:
          - name: config-volume
            mountPath: /etc/statsd
        - name: mixer
          image: "docker.io/istionightly/mixer:circleci-nightly"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 9091
          - containerPort: 9093
          - containerPort: 9094
          - containerPort: 42422
          args:
            - --configStoreURL=k8s://
            - --configDefaultNamespace=istio-system
            - --trace_zipkin_url=http://zipkin:9411/api/v1/spans
            - --logtostderr
          resources:
            {}

        - name: mixer-proxy
          image: "docker.io/istionightly/proxy:circleci-nightly"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 15004
          args:
          - proxy
          - mixer
          - --controlPlaneAuthPolicy
          - NONE #--controlPlaneAuthPolicy
          - --customConfigFile
          - /etc/istio/proxy/envoy_mixer.json
          resources:
            null

          volumeMounts:
          - name: istio-certs
            mountPath: /etc/certs
            readOnly: true
      volumes:
      - name: istio-certs
        secret:
          secretName: istio.istio-mixer-service-account
          optional: true
      - name: config-volume
        configMap:
          name: istio-mixer
```

### New example output

```yaml
---
# Source: istio/charts/mixer/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: istio-mixer
  labels:
    app: mixer
    chart: mixer-0.5.0
    release: test-template
    heritage: Tiller
    istio: mixer
spec:
  ports:
  - name: tcp-plain
    port: 9091
  - name: tcp-mtls
    port: 15004
  - name: http-health
    port: 9093
  - name: configapi
    port: 9094
  - name: statsd-prom
    port: 9102
  - name: statsd-udp
    port: 9125
    protocol: UDP
  - name: prometheus
    port: 42422
  selector:
    istio: mixer

---
# Source: istio/charts/mixer/templates/deployment.yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: istio-mixer
  labels:
    app: mixer
    chart: mixer-0.5.0
    release: test-template
    heritage: Tiller
    istio: mixer
  annotations:
    checksum/config-volume: f9d584221ded283afeff3543a3a561485717933e34482cbc719f6d94a2b8b10a
spec:
  replicas: 1
  template:
    metadata:
      labels:
        istio: mixer
      annotations:
        sidecar.istio.io/inject: "false"
    spec:
      serviceAccountName: istio-mixer-service-account
      containers:
        - name: statsd-to-prometheus
          image: "prom/statsd-exporter:latest"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 9102
          - containerPort: 9125
            protocol: UDP
          args:
          - '-statsd.mapping-config=/etc/statsd/mapping.conf'
          resources:
            {}

          volumeMounts:
          - name: config-volume
            mountPath: /etc/statsd
        - name: mixer
          image: "docker.io/istionightly/mixer:circleci-nightly"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 9091
          - containerPort: 9093
          - containerPort: 9094
          - containerPort: 42422
          args:
            - --configStoreURL=k8s://
            - --configDefaultNamespace=istio-system
            - --trace_zipkin_url=http://zipkin:9411/api/v1/spans
            - --logtostderr
          resources:
            {}

        - name: mixer-proxy
          image: "docker.io/istionightly/proxy:circleci-nightly"
          imagePullPolicy: IfNotPresent
          ports:
          - containerPort: 15004
          args:
          - proxy
          - mixer
          - --controlPlaneAuthPolicy
          - NONE #--controlPlaneAuthPolicy
          - --customConfigFile
          - /etc/istio/proxy/envoy_mixer.json
          resources:
            null

          volumeMounts:
          - name: istio-certs
            mountPath: /etc/certs
            readOnly: true
      volumes:
      - name: istio-certs
        secret:
          secretName: istio.istio-mixer-service-account
          optional: true
      - name: config-volume
        configMap:
          name: istio-mixer

---
```

mandarjog committed:
Use 'helm template' to generate istio artifacts (#4001)

* add splitCheckAndReport option to create separate mixer deployments

* update statsd in deployments

* Update prometheus config map with new names

* update makefile

* Fix submake

* use helm from ISTIO_OUT

* helm should be in ISTIO_OUT

* Update names

* add namespace

* Update statsd sink address

* revert values, no longer needed

* Fix pods running problem for prometheus

* fix istio-mixer-type label

* Use WaitForDeploymentsReady instead of getPods

* revert port names to tcp-*

* add mixer debug

* change mixer to debug

* use global.securityEnabled in Makefile

* Remove prometheus from addons, it is in base helm

* Add e2e-mixer

* Update values

* Create compat updateVersion.sh

* Use mtls as separate variable

* fix scripts

* fix mtls typo

* Set refreshInterval to 1s for tests

* remove debug from Mixer

* Add specific on failure func

* dump config on failure

* Use deployment status instead of pods, which also ensures readiness

* enbale mac local builds

* reapply commit 5be4f6617835189a5dcd7ffd0f56f7aa942c17a2

ping035627 committed:
correct the url of envoy docs (#4498)

Automatic merge from submit-queue.

Correct the url of envoy docs

When I read the code, I found that the URL in the note was not accessible. I modified to improve code readability.

diemtvu committed:
Enforce authentication policy validation. (#4512)

Automatic merge from submit-queue.

Enforce authentication policy validation.

- Fully parse JwksURI to make sure both hostname and port are valid.
- Add addition check to prohibit JWTs with the same issuer.
